src/map/ai/controllers/mob_controller.cpp

```diff
@@ -568,40 +568,60 @@ void CMobController::DoCombatTick(time_point tick)
     HandleEnmity();
     PTarget = static_cast<CBattleEntity*>(PMob->GetEntity(PMob->GetBattleTargetID()));
 
-    if (PTarget)
+    if (TryDeaggro())
     {
-        if (TryDeaggro())
-        {
-            Disengage();
-            return;
-        }
+        Disengage();
+        return;
+    }
 
-        TryLink();
+    TryLink();
```

Comment:
shouldnt this remain inside the PTarget check?

Reply:
At the top of TryLink it does a PTarget nullptr check so this is safe

Reply:
Turns out there is already some logic for following inside https://github.com/LandSandBoat/server/blob/base/scripts/globals/follow.lua

I think it might make the most sense to combine these two since there is some nice logic for handling multiple followers (like we need to with Mother Globe) but it is also missing some functionality such as the runaway. I'm not positive the existing logic works with mobs following players/npcs either but this new follow logic does.